docs/source/advanced/query-mechanics.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Query mechanics
```

Comment:
Do we need this article in the first version?

Reply:
This is a lone-wolf in `advanced` right now too.  My comment with the same perspective: https://github.com/apollographql/apollo-server/pull/942#discussion_r182720602